### PR TITLE
feat(ButtonIcon): implement clickable icon component

### LINKED ISSUE #387: Icons: update clickable icons
Screen Shot 2021-11-16 at 11 03 51

### PR DESCRIPTION
**Pull request by tiagoalvesdulce**
Closes #387 

Screen Shot 2021-11-29 at 14 29 40

### CODE REVIEW

**Review by bgptr — CHANGES_REQUESTED** [review]
Overall looking good, only noticed these issues:
- styles are missing on the disabled state. Cursor, hover and colors stay the same.
- spinner is not visible (white on white background) in loading state:
image

**Comment by tiagoalvesdulce** [comment]
Oh, I completely forgot to implement the disabled state

**Comment by bgptr** [comment]
Looks cool. Just one more thing. The darker icon color should be lighter in the disabled state:
image

File: src/theme/lightTheme.js [hunk 1]
```diff
@@ -214,6 +214,13 @@ const lightTheme = {
   "paginator-selected-color": "#2970FF",
   "paginator-button-color": "#3D5873",
   "paginator-brake-color": "#C4CBD2",
+
+  /** Button Icon */
+  "button-icon-background": "var(--color-white)",
+  "button-icon-color-1": "var(--icon-color)",
+  "button-icon-color-2": "var(--color-primary-dark)",
+  "button-icon-color-1-disabled": "var(--color-gray-light",
```
Comment (bgptr) [inline src/theme/lightTheme.js:222]: missing close parenthesis

**Review by bgptr — COMMENTED** [review]
just one little inline nit, but otherwise LGTM

**Review by bgptr — APPROVED** [review]

**Review by amass01 — APPROVED** [review]
LGTM.

One general  question: why do we need `style` prop if we have `className` ?
I guess for backward compatibility in decrediton ?

**Comment by tiagoalvesdulce** [comment]
@bgptr I'm thinking about ditching the `style` prop to encourage the usage of `className` as per conversation with `amass` on matrix. Do you this is important for Decrediton anyhow?

**Comment by bgptr** [comment]
@tiagoalvesdulce It's not used, go for it.

**Pull request merged**